backend/src/events/views.ts

```diff
@@ -29,89 +40,143 @@ export type EventDTO = {
   };
 };
 
-eventRouter.post("/", async (req: Request, res: Response) => {
-  // #swagger.tags = ['Events']
-  const eventDTO: EventDTO = req.body;
-  attempt(res, 201, () => eventController.createEvent(eventDTO));
-  socketNotify("/events");
-});
-
-eventRouter.put("/:eventid", async (req: Request, res: Response) => {
-  // #swagger.tags = ['Events']
-  attempt(res, 200, () =>
-    eventController.updateEvent(req.params.eventid, req.body)
-  );
-  socketNotify(`/events/${req.params.eventid}`);
-});
-
-eventRouter.delete("/:eventid", async (req: Request, res: Response) => {
-  // #swagger.tags = ['Events']
-  attempt(res, 200, () => eventController.deleteEvent(req.params.eventid));
-  socketNotify("/events");
-});
-
-eventRouter.get("/", async (req: Request, res: Response) => {
-  // #swagger.tags = ['Events']
-  const filter = {
-    date: req.query.date as string,
-    ownerId: req.query.ownerid as string,
-    userId: req.query.userid as string,
-  };
+eventRouter.post(
+  "/",
+  (authIfAdmin as RequestHandler) || (authIfSupervisor as RequestHandler),
```

Comment:
please see my earlier comments. I deally you want something similar to `const useAdminAuth  = authIfAdmin as RequestHandler` at the top so you can re use it